Add docs for new istioctl and operator change (#7791)
istio/istio.io: content/en/docs/setup/upgrade/index.md

The output confirms that the pod is using `istiod-canary` revision of the control plane.

### Uninstall old control plane

After upgrading both the control plane and data plane, you can uninstall the old control plane. For example, the following command uninstalls a control plane of revision `istio-1-6-5`:

{{< text bash >}}
$ istioctl x uninstall --revision istio-1-6-5
{{< /text >}}

Confirm that the old control plane has been removed and only the new one still exists in the cluster:

{{< text bash >}}
$ kubectl get pods -n istio-system -lapp=istiod
NAME                             READY   STATUS    RESTARTS   AGE

> istiod-canary-55887f699c-t8bh8   1/1     Running   0          27m

Reviewer: ```suggestion
istiod-1.7.0-55887f699c-t8bh8   1/1     Running   0          27m
```

{{< /text >}}

> Note that the above instructions only removed the resources for the specified control plane revision, but not cluster-scoped resources shared with other control planes. To uninstall Istio completely, consider using the `--purge` flag. For more information, refer to the [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).

Reviewer: ```suggestion
Note that the above instructions only removed the resources for the specified control plane revision, but not cluster-scoped resources shared with other control planes. To uninstall Istio completely, refer to the [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).```

Author: done

The output confirms that the pod is using `istiod-canary` revision of the control plane.

> ### Uninstall old control plane

Reviewer 2: This needs more explanation of how the different options (-f, --revision, --purge) differ and what we recommend in which situation.

Author: one question is should we put that in the upgrade guide or the istioctl guide

Reviewer: This section is only about upgrade, so it should only talk about uninstalling an old revsion

Author: ok so it would only talk about `istioctl uninstall --revision` in the upgrade doc here, I would add description of other options in the istioctl guide(--purge is added already)

> After upgrading both control plane and data plane, you can uninstall old control plane. For example the following command uninstalls control plane with `revision=default`

Reviewer 3: Grammer.  "After upgrading both the control plane and data plane, you can uninstall the old control plane...."

Author: done

> After upgrading both the control plane and data plane, you can uninstall the old control plane. For example the following command uninstalls control plane with `revision=default`

Author: The `revision called xx` does not sound like a common pattern we have seen for revision, how about revision of `xxx`?

{{< text bash >}}

> $ istioctl x uninstall --revision default

Reviewer 2: best not to use default/canary but rather revision labels.

Author: done

{{< /text >}}

> or if you previously installed Istio with customized installation options, you can run the following command:

Author: done

{{< text bash >}}
$ istioctl x uninstall <your original installation options>
{{< /text >}}

> 

Reviewer: This second option seems confusing. Why is it needed? Aren't the options available in the in-cluster CR?

Author: What I wanted to mention is that users can also uninstall with the generated manifest from the original install option, for example if they previous installed istio with `istioctl install -f iop.yaml --set foo=bar` then `istioctl x uninstall -f iop.yaml --set foo=bar` would remove the generated manifests from the cluster.

Author: To avoid confusion, I moved this part to the ##uninstall istio section in the istioctl guide instead.

> After removal of the old control plane, only the new one still exists in the cluster:

Reviewer: ```suggestion
Confirm that the old control plane has been removed and only the new one still exists in the cluster:
```

{{< text bash >}}
$ kubectl get pods -n istio-system -lapp=istiod
NAME                             READY   STATUS    RESTARTS   AGE
istiod-canary-55887f699c-t8bh8   1/1     Running   0          27m
{{< /text >}}

> Note that the above commands only remove resources owned by specific control plane. To uninstall Istio completely, consider using the `--purge` flag, for more details refer to [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).

Reviewer 4: ```suggestion
Note that the above commands remove the resources for the specified control plane, but not cluster-scoped resources shared between different Istio control planes. To uninstall Istio completely, consider using the `--purge` flag. For more information, refer to the [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).
```

Author: done

> Note that the above commands would only remove resources own by specific control plane, if you want to uninstall Istio completely, consider using `--purge` flag, for more details refer to [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).

Reviewer 3: Grammar.  "Note that the above commands only remove resources owned by specific control planes.  To uninstall Istio completely, consider using the `--purge` flag..."

Author: done

> Note that the above commands remove the resources for the specified control plane, but not cluster-scoped resources shared between different Istio control planes. To uninstall Istio completely, consider using the `--purge` flag. For more information, refer to the [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).

Reviewer: ```suggestion
Note that the above instructions only removed the resources for the specified control plane revision, but not cluster-scoped resources shared with other control planes. To uninstall Istio completely, refer to the [uninstall guide](/docs/setup/install/istioctl/#uninstall-istio).
```
